development/tools/cmd/enablegithubauth/main.go

```diff
@@ -8,102 +8,79 @@ import (
-	envDexGithubIntegrationAppClientID     = "GITHUB_INTEGRATION_APP_CLIENT_ID"
-	envDexGithubIntegrationAppClientSecret = "GITHUB_INTEGRATION_APP_CLIENT_SECRET"
-	envDexCallbackURL                      = "DEX_CALLBACK_URL"
-	envGithubTeamsWithKymaAdmins           = "GITHUB_TEAMS_WITH_KYMA_ADMINS_RIGHTS"
+	clusterUsers = "/resources/core/charts/cluster-users/values.yaml"
+	dexConfigMap = "/resources/dex/templates/dex-config-map.yaml"
 )
 
 func main() {
-	kymaProjectDirVal := os.Getenv(envKymaProjectDir)
-	clientID := os.Getenv(envDexGithubIntegrationAppClientID)
-	clientSecret := os.Getenv(envDexGithubIntegrationAppClientSecret)
-	dexCallbackURL := os.Getenv(envDexCallbackURL)
-	kymaAdmins := os.Getenv(envGithubTeamsWithKymaAdmins)
-
-	if kymaProjectDirVal == "" {
-		log.Fatalf("missing env: %s", envKymaProjectDir)
-	}
-	if clientID == "" {
-		log.Fatalf("missing env: %s", envDexGithubIntegrationAppClientID)
-	}
-	if clientSecret == "" {
-		log.Fatalf("missing env: %s", envDexGithubIntegrationAppClientSecret)
-	}
-	if dexCallbackURL == "" {
-		log.Fatalf("missing env: %s", envDexCallbackURL)
-	}
-	if kymaAdmins == "" {
-		log.Fatalf("missing env: %s", envGithubTeamsWithKymaAdmins)
-	}
+	var cfg Config
+	err := envconfig.Init(&cfg)
+	fatalOnError(err, "while init config")
 
-	kymaPath := fmt.Sprintf("%s/kyma", kymaProjectDirVal)
-	clusterUsers := "/resources/core/charts/cluster-users/values.yaml"
-	dexConfigMap := "/resources/dex/templates/dex-config-map.yaml"
-
-	fUsers, err := os.OpenFile(path.Join(kymaPath, clusterUsers), os.O_RDWR, os.ModeAppend)
-	if err != nil {
-		log.Fatalf("cannot open file %s, %v", path.Join(kymaPath, clusterUsers), err)
-	}
-
-	defer func() {
-		if err := fUsers.Close(); err != nil {
-			panic(err)
-		}
-	}()
+	kymaPath := fmt.Sprintf("%s/kyma", cfg.KymaProjectDir)
+	adjustClusterUsersOrDie(cfg, kymaPath)
+	adjustDexConfigMapOrDie(cfg, kymaPath)
+}
 
-	rClusterUsers := RootClusterUsers{}
-	b, err := ioutil.ReadAll(fUsers)
-	if err != nil {
-		panic(err)
-	}
-	if err := yaml.Unmarshal(b, &rClusterUsers); err != nil {
-		panic(err)
+type (
+	rootClusterUsers struct {
+		Bindings bindings `yaml:"bindings"`
 	}
 
-	teams := strings.Split(kymaAdmins, ",")
-	for _, team := range teams {
-		rClusterUsers.Bindings.KymaAdmin.Groups = append(rClusterUsers.Bindings.KymaAdmin.Groups, fmt.Sprintf("kyma-project:%s", team))
+	bindings struct {
+		KymaAdmin groups `yaml:"kymaAdmin"`
 	}
 
-	n, err := yaml.Marshal(rClusterUsers)
-	if err != nil {
-		panic(err)
+	groups struct {
+		Groups []string `yaml:"groups"`
 	}
 
-	_, err = fUsers.Seek(0, 0)
-	if err != nil {
-		panic(err)
-	}
-	_, err = fUsers.Write(n)
-	if err != nil {
-		panic(err)
-	}
+	rawYAML map[interface{}]interface{}
+)
 
-	fConfigMap, err := os.OpenFile(path.Join(kymaPath, dexConfigMap), os.O_APPEND|os.O_WRONLY, os.ModeAppend)
-	if err != nil {
-		panic(err)
-	}
+func adjustClusterUsersOrDie(cfg Config, kymaPath string) {
```

Comment:
I also had such implementation

```go
func adjustClusterUsersOrDie(cfg Config, kymaPath string) {
	usersFile, err := ioutil.ReadFile(path.Join(kymaPath, clusterUsers))
	if err != nil {
		log.Fatalf("cannot read file %s, %v", path.Join(kymaPath, clusterUsers), err)
	}

	cUsers := rootClusterUsers{}
	err = yaml.Unmarshal(usersFile, &cUsers)
	fatalOnError(err)

	teams := strings.Split(cfg.KymaAdmins, ",")
	for _, team := range teams {
		cUsers.Bindings.KymaAdmin.Groups = append(cUsers.Bindings.KymaAdmin.Groups, fmt.Sprintf("kyma-project:%s", team))
	}

	overrides := &v1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name: "cluster-users-oauth-override",
			Labels: map[string]string{
				"kyma-project.io/installation": "",
				"component":                    "core",
				"installer":                    "overrides",
			},
		},
		Data: map[string]string{
			"cluster-users.bindings.kymaAdmin.groups": fmt.Sprintf("%+q", cUsers.Bindings.KymaAdmin.Groups),
		},
	}

	coreClient := corev1.NewForConfigOrDie(config.GetConfigOrDie())
	_, err = coreClient.ConfigMaps("kyma-installer").Create(overrides)
	fatalOnError(err)
}
```

In this case, I created a ConfigMap with overrides for `groups` entry. Unfortunately, the `groups` entry is an array and installer right now does not have any support for that. 

Overrides are done by ConfigMaps data which are `map[string]string`. In installer, it is also read as [`map[string]string`](https://github.com/kyma-project/kyma/blob/master/components/installer/pkg/overrides/config-reader.go#L25-L26)

Then such overrides are propagated to Helm Chart and in values.yaml instead of 
```
bindings:
  kymaAdmin:
    groups: ['kyma-project:cluster-access']
```
we have
```
bindings:
  kymaAdmin:
    groups: "['kyma-project:cluster-access']"
```
which causing error: `range can't iterate over ['kyma-project:cluster-access']`